modules/UI/authentication/AuthHandler.js

```diff
@@ -1,227 +0,0 @@
-/* global APP */
-import Logger from '@jitsi/logger';
-
-import { openConnection } from '../../../connection';
-import {
-    openAuthDialog,
-    openLoginDialog } from '../../../react/features/authentication/actions.web';
-import {
-    LoginDialog,
-    WaitForOwnerDialog
-} from '../../../react/features/authentication/components';
-import {
-    getTokenAuthUrl,
-    isTokenAuthEnabled
-} from '../../../react/features/authentication/functions';
-import { getReplaceParticipant } from '../../../react/features/base/config/functions';
-import { isDialogOpen } from '../../../react/features/base/dialog/functions';
-import { setJWT } from '../../../react/features/base/jwt/actions';
-import UIUtil from '../util/UIUtil';
-
-import ExternalLoginDialog from './LoginDialog';
-
-
-let externalAuthWindow;
-
-const logger = Logger.getLogger(__filename);
-
-
-/**
- * Authenticate using external service or just focus
- * external auth window if there is one already.
- *
- * @param {JitsiConference} room
- * @param {string} [lockPassword] password to use if the conference is locked
- */
-function doExternalAuth(room, lockPassword) {
-    const config = APP.store.getState()['features/base/config'];
-
-    if (externalAuthWindow) {
-        externalAuthWindow.focus();
-
-        return;
-    }
-
-    if (room.isJoined()) {
-        let getUrl;
-
-        if (isTokenAuthEnabled(config)) {
-            getUrl = Promise.resolve(getTokenAuthUrl(config)(room.getName(), true));
-            initJWTTokenListener(room);
-        } else {
-            getUrl = room.getExternalAuthUrl(true);
-        }
-        getUrl.then(url => {
-            externalAuthWindow = ExternalLoginDialog.showExternalAuthDialog(
-                url,
-                () => {
-                    externalAuthWindow = null;
-                    if (!isTokenAuthEnabled(config)) {
-                        room.join(lockPassword);
-                    }
-                }
-            );
-        });
-    } else if (isTokenAuthEnabled(config)) {
-        redirectToTokenAuthService(room.getName());
-    } else {
-        room.getExternalAuthUrl().then(UIUtil.redirect);
-    }
-}
-
-/**
- * Redirect the user to the token authentication service for the login to be
- * performed. Once complete it is expected that the service will bring the user
- * back with "?jwt={the JWT token}" query parameter added.
- * @param {string} [roomName] the name of the conference room.
- */
-export function redirectToTokenAuthService(roomName) {
-    const config = APP.store.getState()['features/base/config'];
-
-    // FIXME: This method will not preserve the other URL params that were
-    // originally passed.
-    UIUtil.redirect(getTokenAuthUrl(config)(roomName, false));
```

Comment:
I think we can remove `redirect` from UIUtil.

Reply:
I thought I did remove it hum ...

Reply:
done.